bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/watson/ElementTreeWriter.java

```diff
@@ -78,65 +79,23 @@ public Object readData(IPath path, DataInput input) {
 	}
 
 	/**
-	 * Sorts the given array of trees so that the following rules are true:
-	 * 	 - The first tree has no parent
-	 * 	 - No tree has an ancestor with a greater index in the array.
-	 * If there are no missing parents in the given trees array, this means
-	 * that in the resulting array, the i'th tree's parent will be tree i-1.
-	 * The input tree array may contain duplicate trees.
-	 * The sort order is written to the given output stream.
+	 * Sorts the given array of trees and The sort order is written to the given
+	 * output stream.
 	 */
 	protected ElementTree[] sortTrees(ElementTree[] trees, DataOutput output) throws IOException {
+		ElementTree[] sorted = SaveManager.sortTrees(trees);
```

Comment:
`sortTrees()` can return null. The sort that was before here would throw IOException in that case, now NP exception. This is a problem, because the two callers of `org.eclipse.core.internal.watson.ElementTreeWriter.writeDeltaChain(ElementTree[], IPath, int, DataOutput, IElementComparator)` have dedicated catches for IO exception but nothing for NP.

That must be fixed.